src/parser/alter.rs

```diff
@@ -39,6 +39,15 @@ impl Parser<'_> {
         ))
     }
 
+    pub fn parse_alter_user(&mut self) -> Result<Statement, ParserError> {
+        if dialect_of!(self is PostgreSqlDialect) {
+            return self.parse_pg_alter_user();
+        }
```

Comment:
I think we can skip the dialect checks and leave the parser to always accept the statement if it shows up